lua/nvim_lsp/sumneko_lua.lua

```diff
@@ -0,0 +1,104 @@
+local skeleton = require 'nvim_lsp/skeleton'
+local util = require 'nvim_lsp/util'
+local vim = vim
+
+local name = "sumneko_lua"
+
+local function make_installer()
+	local P = util.path.join
+  local install_dir = P{util.base_install_dir, name}
+  local git_dir = P{install_dir, "lua-language-server"}
+
+  local bin = P{git_dir, "bin", "Linux", "lua-language-server"}
+  local main_file = P{git_dir, "main.lua"}
+	local cmd = {bin, '-E', main_file}
+
+	local X = {}
+	function X.install()
+		local install_info = X.info()
+		if install_info.is_installed then
+			print(name, "is already installed")
+			return
+		end
+		if not (util.has_bins("ninja") or util.has_bins("curl")) then
+			error('Need either "ninja" or "curl" (to download ninja) to install this.')
+			return
+		end
+		if not util.has_bins("sh", "chmod", "unzip") then
+			error('Need the binaries "sh", "chmod", "unzip" to install this')
+			return
+		end
+		local script = [=[
+set -e
+# Install ninja if not available.
+which ninja >/dev/null || {
+	test -x ninja || {
+		curl -LO https://github.com/ninja-build/ninja/releases/download/v1.9.0/ninja-linux.zip
```

Comment:
Other platforms? windows, macos

Reply:
@zhaozg I could do mac and I guess windows has its subsystem. I'll consider adding those.